Source/WebCore/platform/graphics/ImageFrameProvider.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2024 Apple Inc.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "ImageFrame.h"
+#include "ImageFrameWorkQueue.h"
+#include "ImageTypes.h"
+#include <wtf/ThreadSafeWeakPtr.h>
+
+namespace WebCore {
+
+class BitmapImageSource;
+class ImageDecoder;
+class ImageFrame;
+class NativeImage;
+
+class ImageFrameProvider : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<ImageFrameProvider> {
+public:
+    static Ref<ImageFrameProvider> create(BitmapImageSource&, Ref<ImageDecoder>&&);
+    ~ImageFrameProvider();
+
+    Ref<BitmapImageSource> protectedSource() const { return m_source.get().releaseNonNull(); }
+    Ref<ImageDecoder> protectedDecoder() const { return m_decoder; }
+
+    // Encoded and decoded data
+    void encodedDataStatusChanged(EncodedDataStatus);
+    void destroyDecodedData(bool destroyAll);
+    void decodePropertiesChanged(unsigned decodedPropertiesSize);
```

Comment:
Should this be called `decodedPropertiesChanged`?

Reply:
Name changed to `decodedPropertiesSizeChanged()` which is similar to `decodedSizeIncreased()` and `decodedSizeDecreased()`.